tools/utils/utils.js

```diff
@@ -490,7 +490,7 @@ exports.isUrlWithSha = function (x) {
 exports.isNpmUrl = function (x) {
   // These are the various protocols that NPM supports, which we use to download NPM dependencies
   // See https://docs.npmjs.com/files/package.json#git-urls-as-dependencies
-  return exports.isUrlWithSha(x) || /^(git|git\+ssh|git\+http|git\+https)?:\/\//.test(x);
+  return /^(git|git\+ssh|git\+http|git\+https|https|http)?:\/\//.test(x);
```

Comment:
Would you mind adding some tests for `isNpmUrl` to verify the accepted dependency URL formats? They don't have to be that extensive, just one quick check for each accepted URL format, including `http` and `https`. The tests would go in [`tools/tests/utils-tests.js`](https://github.com/meteor/meteor/blob/1aaeca1cb82f080e15267b1dc4dc724074dc3b1a/tools/tests/utils-tests.js).

Reply:
I would also like to be sure that these new URLs are preserved correctly in `npm-shrinkwrap.json`, so that we don't keep trying to reinstall the packages when they've already been installed.

Reply:
I have added a few tests.

> I would also like to be sure that these new URLs are preserved correctly in npm-shrinkwrap.json, so that we don't keep trying to reinstall the packages when they've already been installed.

Does this comment have to do with this MR ?